xl/playlist.py

```diff
@@ -2209,18 +2209,17 @@ def save_to_location(self, location):
             used to restore their order
         """
         if os.path.exists(location):
-            f = open(location + ".new", "w")
+            destination = location + ".new"
         else:
-            f = open(location, "w")
-        for playlist in self.playlists:
-            f.write(playlist)
-            f.write('\n')
+            destination = location
+        with open(destination, 'w') as f:
+            for playlist in self.playlists:
+                f.write(playlist)
+                f.write('\n')
+            f.write("EOF\n")
```

Comment:
These can be `print(..., file=f)` without the `\n`, if you prefer. Doesn't matter though.